Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1920,13 +1920,33 @@ extension BrowserViewController: WKUIDelegate {
         // If the page we just opened has a bad scheme, we return nil here so that JavaScript does not
         // get a reference to it which it can return from window.open() - this will end up as a
         // CFErrorHTTPBadURL being presented.
-        guard let scheme = (navigationAction.request as NSURLRequest).url?.scheme?.lowercased(), SchemesAllowedToOpenPopups.contains(scheme) else {
-            return nil
+        if #available(iOS 11, *) {
+            guard canRequestBeOpenedAsPopup(request) else {
+                return nil
+            }
+        } else {
+            // Workaround for iOS 10 where `window.open()` for "about:blank"
+            // creates a request with an empty URL (no scheme).
+            if request.url?.absoluteString == "" {
+                return newTab.webView
+            }
+
+            guard canRequestBeOpenedAsPopup(request) else {
+                return nil
+            }
         }
 
         return newTab.webView
     }
 
+    fileprivate func canRequestBeOpenedAsPopup(_ request: URLRequest) -> Bool {
```

Comment:
Nit: naming. Having spent a little while with UIKit protocols, I'm beginning to want to see `can` methods as `should` methods.